docs/services/rook-services-guide.md

```diff
@@ -0,0 +1,800 @@
+   section][gcp-stack-install] of the install guide for more information.)
+
+```bash
+cat > stack-gcp.yaml <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: gcp
+---
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: stack-gcp
+  namespace: gcp
+spec:
+  package: "crossplane/stack-gcp:master"
+EOF
+
+kubectl apply -f stack-gcp.yaml
+```
+
+3. Install the Rook stack into Crossplane (See the [Rook stack
+   section][rook-stack-install] of the install guide for more information.)
+
+```bash
+cat > stack-rook.yaml <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: rook
+---
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: stack-rook
+  namespace: rook
+spec:
+  package: "crossplane/stack-rook:master"
+EOF
+
+kubectl apply -f stack-rook.yaml
+```
+
+4. Obtain GCP credentials. (See the [Cloud Provider Credentials][cloud-creds]
+   docs for more information.)
+
+#### GCP Provider
+
+Using the service account json `crossplane-gcp-provider-key.json` that you
+acquired from GCP:
+
+* Generate Base64 encoded value to store in a `Secret`:
+
+  ```bash
+  export BASE64ENCODED_GCP_PROVIDER_CREDS=$(base64 crossplane-gcp-provider-key.json | tr -d "\n")
+  ```
+
+  * Define a GCP `Provider` (using the `PROJECT_ID` environment variable we set
+    above) and `Secret`:
+
+    ```bash
+    cat > gcp-provider.yaml <<EOF
+    ---
+    apiVersion: v1
+    data:
+      credentials.json: $BASE64ENCODED_GCP_PROVIDER_CREDS
+    kind: Secret
+    metadata:
+      name: gcp-provider-creds
+      namespace: crossplane-system
+    type: Opaque
+    ---
+    apiVersion: gcp.crossplane.io/v1alpha2
+    kind: Provider
+    metadata:
+      name: gcp-provider
+    spec:
+      credentialsSecretRef:
+        name: gcp-provider-creds
+        namespace: crossplane-system
+        key: credentials.json
+      projectID: $PROJECT_ID
+    EOF
+
+    kubectl apply -f gcp-provider.yaml
+    unset BASE64ENCODED_GCP_PROVIDER_CREDS # we don't need this anymore.
+    ```
+
+* Verify GCP provider was successfully registered by the crossplane
+
+  ```bash
+  kubectl get providers.gcp.crossplane.io
+  kubectl -n crossplane-system get secrets
+  ```
+
+#### Rook Provider
+
+Rook differs from traditional cloud provider stacks in that it does not come
+with a Rook `Provider` custom resource. The reason for this is that Rook runs in
+any Kubernetes cluster. Therefore, it utilizes the general Kubernetes `Provider`
+custom resource that is packaged as part of the core Crossplane installation. We
+will see how to use this `Provider` type further along in this guide.
+
+## Set Up Resource Classes
+
+To keep all configuration organized in a single location, create a new
+directory:
+
+```bash
+mkdir yugastore && cd $_
+```
+
+> **TL;DR**: if you want to skip the rest of the guide and just deploy
+> Yugastore, you can run `kubectl apply -f <TODO: INSERT URL HERE>`
```

Comment:
@jbw976 where do we want to host the files?

Reply:
i've actually really liked the `kubectl apply -k <url>` functionality that is available broadly in `kubectl` now.  You can point it at a remote folder, for example on github.  I looked into this a lot last night and that seems to be the officially supported way to use `kubectl` to apply an entire remote directory.

I'd say we can just put those files into an examples folder in this repo.

One caveat is that since it's using `kustomize` under the covers, the folder will need to have a `kustomization.yaml` file, but that can be **very** simple and just list a wildcard pattern for which files to include (e.g. `folder/*`).

One fairly complete example of this that @soorena776 and I worked through together is shown below:
```
kubectl apply -k github.com/soorena776/crossplane//cluster/examples/workloads/kubernetes/wordpress/aws/gitops/infra-aws?ref=issue_948
```

Notes:
* notice the double forward slash syntax, that's what separates the repo from the subfolder within the repo
* `?ref=issue_948` is the branch name or it could be a version tag such as `v0.1.0`.  If this param is left off the URL then the default branch is used, e.g. `master`.
* see the `kustomization.yaml` file in the folder at https://github.com/soorena776/crossplane/blob/issue_948/cluster/examples/workloads/kubernetes/wordpress/aws/gitops/infra-aws/kustomization.yaml.  It is listing all files, but I think it could just use wildcards instead.  It doesn't have any crazy patches or other functionality that complicates things.

Put all that together and I believe you get a fairly uncomplicated experience for the user to simply kubectl apply an entire remote folder.  Let me know if you have any questions.

Reply:
Awesome, thank you!